static/js/components/forms/CouponForm.js

```diff
@@ -236,6 +241,27 @@ export const CouponForm = ({
           </div>
         </div>
         <div className="flex">
+          <label htmlFor="is_global">
+            <Field
+              type="checkbox"
+              name="is_global"
+              checked={values.is_global}
+              onChange={() => {
+                values.is_global = !values.is_global
+                setFieldValue("is_global", values.is_global)
+                if (values.is_global) {
+                  setFieldValue("discount", 100)
```

Comment:
Instead of intentionally changing the user entered value to `100%` while clicking on `Global Check, I still prefer this as suggested before: 

-   Global Coupon field should be disabled/hidden when the discount is not 100%.
-   Global Coupon check-field option should be enabled/appeared when user enters the discount of 100%. 
-   You can keep the discount field disabled until the Global field is unchecked.

Reply:
Updated, in light of the AC update this is no longer needed. Checking `is_global` now only hides and empties the product selection.